File: app/src/main/java/com/rodolfonavalon/canadatransit/model/database/OperatorFeedVersion.java
package com.rodolfonavalon.canadatransit.model.database;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import lombok.Getter;

public class OperatorFeedVersion {

    @SerializedName("sha1")
    @Getter private String sha1;
    @SerializedName("earliest_calendar_date")
    @Getter private String earliestCalendarDate;
    @SerializedName("latest_calendar_date")
    @Getter private String latestCalendarDate;
    @SerializedName("md5")
    @Getter private String md5;
    @SerializedName("fetched_at")
    @Getter private String fetchedAt;
    @SerializedName("imported_at")
    @Getter private String importedAt;
    @SerializedName("created_at")
    @Getter private String createdAt;
    @SerializedName("updated_at")
    @Getter private String updatedAt;
    @SerializedName("feed_version_imports_url")
    @Getter private String feedVersionImportsUrl;
    @SerializedName("import_status")
    @Getter private String importStatus;
    @SerializedName("feed")
    @Getter private String feed;
    @SerializedName("url")
    @Getter private String url;
    @SerializedName("download_url")
    @Getter private String downloadUrl;
    @SerializedName("feedvalidator_url")
    @Getter private String feedValidatorUrl;

    @SerializedName("import_level")
    @Getter private int importLevel;

    @SerializedName("is_active_feed_version")
    @Getter private boolean isActiveFeedVersion;

    @SerializedName("feed_version_infos")
    @Getter private List<Integer> feedVersionInfos;
    @SerializedName("feed_version_imports"  )
    @Getter private List<Integer> feedVersionImports;
    @SerializedName("changesets_imported_from_this_feed_version")
    @Getter private List<Integer> changesetImportedFromThisFeedVersion;

}

File: app/src/main/kotlin/com/rodolfonavalon/canadatransit/model/database/OperatorFeedVersion.kt
package com.rodolfonavalon.canadatransit.model.database

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverters
import com.google.gson.annotations.SerializedName
import com.rodolfonavalon.canadatransit.model.database.converter.room.TransitLandConverter
import org.joda.time.DateTime

@Entity
@TypeConverters(TransitLandConverter::class)
class OperatorFeedVersion(
        @PrimaryKey
        @SerializedName("feed") val feedOneStopId: String,

        @SerializedName("sha1") val sha1: String,
        @SerializedName("earliest_calendar_date") val earliestCalendarDate: DateTime,
        @SerializedName("latest_calendar_date") val latestCalendarDate: DateTime,
        @SerializedName("md5") val md5: String,
        @SerializedName("fetched_at") val fetchedAt: DateTime,
        @SerializedName("imported_at") val importedAt: DateTime,
        @SerializedName("created_at") val createdAt: DateTime,
        @SerializedName("updated_at") val updatedAt: DateTime,
        @SerializedName("feed_version_imports_url") val feedVersionImportsUrl: String,
        @SerializedName("import_status") val importStatus: String,
        @SerializedName("url") val url: String,
        @SerializedName("download_url") val downloadUrl: String,
        @SerializedName("feedvalidator_url") val feedValidatorUrl: String,
        @SerializedName("import_level") val importLevel: Int,
        @SerializedName("is_active_feed_version") val isActiveFeedVersion: Boolean,
        @SerializedName("feed_version_infos") val feedVersionInfos: List<Int>,
        @SerializedName("feed_version_imports") val feedVersionImports: List<Int>,
        @SerializedName("changesets_imported_from_this_feed_version") val changesetImportedFromThisFeedVersion: List<Int>
)
